### PR TITLE
feat(question-type-cursor): #FOR-498 sync question specific field's in eventBus

### PR DESCRIPTION
**Pull request by c4role**
## Describe your changes
Sync specific field's in eventBus for 'formulaire-public' + refacto

## Checklist tests
Create a new public form with at least a question type cursor, share it and check if input type range is correct (specificly the blue part)

## Issue ticket number and link
[FOR-498](https://jira.support-ent.fr/browse/FOR-498)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

**Review by alicedraillard — APPROVED** [review]

File: formulaire/src/test/java/fr/openent/formulaire/service/test/impl/DefaultQuestionServiceTest.java [hunk 1]
```diff
@@ -80,7 +80,7 @@ public void testListForFormAndSection(TestContext ctx) {
             ctx.assertEquals(expectedParams.toString(), body.getJsonArray(VALUES).toString());
             async.complete();
         });
-        defaultQuestionService.listForFormAndSection(FORM_ID, null);
+        defaultQuestionService.listForFormAndSection(FORM_ID);
```
Comment (qperie) [inline formulaire/src/test/java/fr/openent/formulaire/service/test/impl/DefaultQuestionServiceTest.java:83]: Les tests de promise sont pas exactement comme les tests de handler, voilà un exemple : https://github.com/OPEN-ENT-NG/form/pull/171/files#diff-a5cfb3e7320d43c298ca02ad201e5dc771d28ca88ae382cf18f54500a86b6c0c

File: formulaire/src/test/java/fr/openent/formulaire/service/test/impl/DefaultQuestionSpecificServiceTest.java [hunk 2]
```diff
@@ -43,7 +43,7 @@ public void testListSpecificField(TestContext ctx) {
             ctx.assertEquals(expectedParams.toString(), body.getJsonArray(VALUES).toString());
             async.complete();
         });
-        defaultQuestionSpecificFieldService.listByIds(questionIds, null);
+        defaultQuestionSpecificFieldService.listByIds(questionIds);
```
Comment (qperie) [inline formulaire/src/test/java/fr/openent/formulaire/service/test/impl/DefaultQuestionSpecificServiceTest.java:46]: cf au-dessus

**Review by qperie — CHANGES_REQUESTED** [review]

**Review by qpic-cgi — APPROVED** [review]

File: formulaire/src/main/java/fr/openent/formulaire/service/impl/DefaultQuestionSpecificFieldService.java [hunk 3]
```diff
@@ -3,23 +3,49 @@
...
 import org.entcore.common.sql.SqlStatementsBuilder;
 
 import static fr.openent.form.core.constants.Fields.*;
 import static fr.openent.form.core.constants.Tables.*;
+import static fr.openent.form.helpers.UtilsHelper.getIds;
 
 public class DefaultQuestionSpecificFieldService implements QuestionSpecificFieldService {
     private final Sql sql = Sql.getInstance();
 
+    private static final Logger log = LoggerFactory.getLogger(QuestionController.class);
```
Comment (qperie) [inline formulaire/src/main/java/fr/openent/formulaire/service/impl/DefaultQuestionSpecificFieldService.java:28]: Attention à la class passée pour le logger

**Review by qperie — APPROVED** [review]
Juste le logger à changer mais impec' pour le reste 👍

**Pull request merged**